src/sentry_logs.c

```diff
@@ -4,9 +4,237 @@
 #include "sentry_options.h"
 #include "sentry_os.h"
 #include "sentry_scope.h"
+#include "sentry_sync.h"
 
 #include <stdarg.h>
 #include <string.h>
+#include <unistd.h>
+
+// TODO think about this
+#ifdef SENTRY_UNITTEST
+#    define QUEUE_LENGTH 5
+#else
+#    define QUEUE_LENGTH 100
+#endif
+#define FLUSH_TIMER 5
+
+typedef struct {
+    sentry_value_t logs[QUEUE_LENGTH];
+    volatile long index;
+    volatile long adding;
+} log_single_buffer_t;
+
+// TODO look at mutex init for inspiration on how to dynamically init
+//  logs as sentry_value_t list using sentry_value_new_list_with_size(...)
+//  (at runtime since function call is not compile-time available)
+static struct {
+    log_single_buffer_t queue;
+    sentry_bgworker_t *timer_worker;
+    volatile long timer_task_submitted;
+    volatile long timer_running;
+    volatile long flushing;
+    sentry_cond_t request_flush;
+    sentry_cond_t adding_done;
+    sentry_mutex_t adding_lock;
```

Comment:
Why are those two necessary? The `adding` topic is entirely an atomic concern. Let's reduce the number of sync variables to a minimum. Adding the condvar + mutex on top of an atomic counter will lead to all kinds of complexities and is unnecessary, since the flusher already spins on the counter (i.e., the counter acts as a signal).